src/Common/src/Interop/Windows/SChannel/Interop.Sec_Application_Protocols.cs

```diff
@@ -41,21 +41,21 @@ public static unsafe byte[] ToByteArray(IList<SslApplicationProtocol> applicatio
             protocols.ProtocolExtenstionType = ApplicationProtocolNegotiationExt.ALPN;
             protocols.ProtocolListSize = (short)protocolListSize;
 
+            Span<byte> pBuffer = new byte[protocolListSize];
+            int index = 0;
+            for (int i = 0; i < applicationProtocols.Count; i++)
+            {
+                pBuffer[index++] = (byte)applicationProtocols[i].Protocol.Length;
```

Comment:
`Debug.Assert(applicationProtocols[i].Protocol.Length < 255)`?  (or `<=`, if appropriate)

Reply:
This check is done, in the code just before in the same function, so a debug assert is not required here.